File: leetcode/dynamic/climbing_stairs.py
def climbStairs(self, n: int) -> int:
    if n <= 2:
        return n
    else:
        a = 1
        b = 2
        s = 0
        n -= 2
        while n:
            s = a + b
            a = b
            b = s
            n -= 1
        return s

File: leetcode/dynamic/climbing_stairs.cpp
int climbStairs(int n) {
	if (n < 4) {
		return n;
	}
	int a, b, s;
	s = 0;
	a = 1; 
	b = 2;
	n = n - 2;
	while (n) {
		s = a + b;
		a = b;
		b = s;
		n--;
	}
	return s;
}
